beacon-chain/sync/initial-sync/service.go

```diff
@@ -154,6 +142,40 @@ func (s *InitialSync) Syncing() bool {
 	return !s.synced
 }
 
+func (s *InitialSync) Resync() error {
+	// set it to false since we are syncing again
+	s.synced = false
+	headState, err := s.chain.HeadState(context.Background())
+	if err != nil {
+		return errors.Wrap(err, "could not retrieve head state")
+	}
+	genesis := time.Unix(int64(headState.GenesisTime), 0)
+
+	s.waitForMinimumPeers()
+	if err := s.roundRobinSync(genesis); err != nil {
+		return errors.Wrap(err, "could not retrieve head state")
+	}
+	log.Infof("Synced up to slot %d", s.chain.HeadSlot())
+
+	s.synced = true
+	return nil
+}
+
+func (s *InitialSync) waitForMinimumPeers() {
+	// Every 5 sec, report handshake count.
+	for {
+		count := len(s.p2p.Peers().Connected())
+		if count >= minStatusCount {
+			break
+		}
+		log.WithField(
+			"handshakes",
+			fmt.Sprintf("%d/%d", count, minStatusCount),
```

Comment:
This line has always made me twitch.  As you're there, what do you think about breaking these out in to two separate fields?

Reply:
alright done